File: backend/Instagram/Instagram/Models/Post.cs
using System;

namespace Instagram
{
    public class Post
    {
        public int Id { get; set; }

        public string link { get; set; }

        public string caption { get; set; }

        public int likeCount { get; set; }

        public int commentCount { get; set; }
        
        public string location { get; set; }

        public int commentStatus { get; set; }

        public int likeCountStatus { get; set; }
        public DateTime timeStamp { get; set; }
        public User user { get; set; }
    }
}

File: frontend/src/app/core/interfaces/post/post.ts
import {Comment} from "../react/comment";
import { Like } from "../react/like";
import { User } from "../user/user";

export interface Post {
  postId: string;
  user:User
  link: string;
  caption: string;
  timeStamp: string;
  likes: number;
  comments: number;
  profileLink?: string;
  liked?: Like;
  isLiked?: boolean;
}
